akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -324,7 +341,9 @@ private[http2] abstract class Http2Demux(http2Settings: Http2CommonSettings, ini
               }
             case PingFrame(false, data) =>
               multiplexer.pushControlFrame(PingFrame(ack = true, data))
-
+            case GoAwayFrame(lastStreamId, _, _) =>
+              // TODO do we log the error code and debug message on some level?
```

Comment:
Doesn't it surface in the user level stream as a failure somehow?

Reply:
Not currently, at least.

Reply:
I guess in general it shouldn't happen when a request is in progress though, but let requests complete, so maybe logging something is more reasonable.